feat: add terraform for oci oke

BobyMCbobs opened the pull request:
terraform to provision an OKE cluster on OCI

BobyMCbobs committed:
feat: add terraform for oci oke

terraform to provision an OKE cluster on OCI

BobyMCbobs commented:
Destroy appears to fail to clean up
- oci_core_internet_gateway
- oci_core_route_table
- oci_core_security_list
- oci_core_subnet
- oci_core_vcn

<img width="1494" alt="Screenshot 2024-08-30 at 09 55 09" src="https://github.com/user-attachments/assets/2a9bc6ad-d728-4a87-8c7d-d3c513a89418">

BobyMCbobs commented:
Error resulting with the following message

```
│ Error: 409-Conflict, The Subnet [REDACTED] references the service VNIC [REDACTED]. You must remove the reference to proceed with this operation.
│ Suggestion: The resource is in a conflicted state. Please retry again or contact support for help with service: Core Subnet
│ Documentation: https://registry.terraform.io/providers/oracle/oci/latest/docs/resources/core_subnet
│ API Reference:
│ Request Target: DELETE [REDACTED]
│ Provider version: 6.7.0, released on 2024-08-17. This provider is 1 Update(s) behind to current.
│ Service: Core Subnet
│ Operation Name: DeleteSubnet
│ OPC request ID: 7072042662d56809ae9d17a2e3619241/4DE3E488FA9E1BCD4DD813FAE463F584/F4E785041C199C19BFAE79A85E980E0F
```

hh commented:
Might want to open an issue with the terraform provider for oracle if you think it's a bug. First glance I'd think that's the case.

BobyMCbobs commented:
Provisioning takes 20mins.

BobyMCbobs commented:
Opened a ticket here
https://github.com/oracle/terraform-provider-oci/issues/2185

BobyMCbobs committed:
feat: integrate into main and start from scratch

- use oke module in main
- remove currently not used terraform (reintroduce when needed)

BobyMCbobs committed:
chore: add kubeconfig output for oke cluster

adds output to get kubeconfig

BobyMCbobs committed:
feat: update docs

use latest info

BobyMCbobs committed:
chore: use kubernetes tfstate backend

use an endpoint which everyone can access

BobyMCbobs committed:
chore: update docs

add auth and tfstate info

BobyMCbobs committed:
chore: update formatting

change expression
